src/server/stat_job.c

```diff
@@ -378,12 +378,16 @@ status_subjob(job *pjob, struct batch_request *preq, svrattrl *pal, int subj, pb
 	if ((pjob->ji_qs.ji_svrflags & JOB_SVFLG_ArrayJob) == 0)
 		return PBSE_IVALREQ;
 
-	/* if subjob is running, use real job structure */
-
-	if (get_subjob_state(pjob, subj) == JOB_STATE_RUNNING) {
-		psubjob = find_job(mk_subjob_id(pjob, subj));
-		if (psubjob)
-			status_job(psubjob, preq, pal, pstathd, bad);
+	/* if subjob job obj exists, use real job structure */
+
```

Comment:
Update the comment to tell why for subjobs state is being updated back to JOB_STATE_FINISHED from JOB_SUBSTATE_EXPIRED in case of job history enabled.

Reply:
Fixed in rc 17 https://github.com/PBSPro/pbspro/pull/590/commits/f94a3405ef13edaa87df40a89af36420b9627c7b